tests/components/test_conversation.py

```diff
@@ -187,9 +203,38 @@ def test_turn_off_intent(hass, sentence):
     assert call.data == {'entity_id': 'light.kitchen'}
 
 
+@asyncio.coroutine
+@pytest.mark.parametrize('sentence', ('toggle kitchen', 'kitchen toggle'))
+def test_toggle_intent(hass, sentence):
```

Comment:
The tests for the generic intents need to be moved to `tests.components.test_init`.

Reply:
And then they shouldn't depend on conversation.

Reply:
I can add test_init but I think these tests should remain here since they are also testing the utterance matching and such.

Reply:
[`components/test_init.py` already exists.](https://github.com/home-assistant/home-assistant/blob/dev/tests/components/test_init.py)

Testing the utterances is good, but we also need to add tests to the other file as that's where they are defined. We can't have core logic be only tested inside component tests.